lib/galaxy/objectstore/irods.py

```diff
@@ -0,0 +1,668 @@
+    exists that is used as an intermediate location for files between Galaxy and iRODS.
+    """
+
+    store_type = 'irods'
+
+    def __init__(self, config, config_dict):
+        super(IRODSObjectStore, self).__init__(config, config_dict)
+
+        auth_dict = config_dict.get('auth')
+        if auth_dict is None:
+            _config_dict_error('auth')
+
+        self.username = auth_dict.get('username')
+        if self.username is None:
+            _config_dict_error('auth->username')
+        self.password = auth_dict.get('password')
+        if self.password is None:
+            _config_dict_error('auth->password')
+
+        resource_dict = config_dict['resource']
+        if resource_dict is None:
+            _config_dict_error('resource')
+        self.resource = resource_dict.get('name')
+        if self.resource is None:
+            _config_dict_error('resource->name')
+
+        zone_dict = config_dict['zone']
+        if zone_dict is None:
+            _config_dict_error('zone')
+        self.zone = zone_dict.get('name')
+        if self.zone is None:
+            _config_dict_error('zone->name')
+
+        connection_dict = config_dict['connection']
+        if connection_dict is None:
+            _config_dict_error('connection')
+        self.host = connection_dict.get('host')
+        if self.host is None:
+            _config_dict_error('connection->host')
+        self.port = connection_dict.get('port')
+        if self.port is None:
+            _config_dict_error('connection->port')
+
+        cache_dict = config_dict['cache']
+        if cache_dict is None:
+            _config_dict_error('cache')
+        self.cache_size = cache_dict.get('size', -1)
+        if self.cache_size is None:
+            _config_dict_error('cache->size')
+        self.staging_path = cache_dict.get('path') or self.config.object_store_cache_path
+        if self.staging_path is None:
+            _config_dict_error('cache->path')
+
+        extra_dirs = dict((e['type'], e['path']) for e in config_dict.get('extra_dirs', []))
+        if not extra_dirs:
+            _config_dict_error('extra_dirs')
+        self.extra_dirs.update(extra_dirs)
+
+        self._initialize()
+
+    def __del__(self):
+        self.session.cleanup()
+
+    def _initialize(self):
+        if irods is None:
+            raise Exception(IRODS_IMPORT_MESSAGE)
+
+        self.home = "/" + self.zone + "/home/" + self.username
+
+        self.session = self._configure_connection(host=self.host, port=self.port, user=self.username, password=self.password, zone=self.zone)
+
+    def _configure_connection(self, host='localhost', port='1247', user='rods', password='rods', zone='tempZone'):
+        with iRODSSession(host=host, port=port, user=user, password=password, zone=zone) as session:
+            # Throws NetworkException if connection fails
+            try:
+                session.pool.get_connection()
```

Comment:
Is there a timeout option here ? This seems to hang forever if IRODS isn't quite up yet.

Reply:
Added timeout to config.